### DISCUSSION TITLE
How can a user set a priority when submitting jobs through the API

### DISCUSSION THREAD

**Discussion by marianne013** [DIRACGrid/DIRAC#6841] [Q&A] [answered]
Hi All,

We have a user who wants to prioritize some of his jobs over others. I can see a UserPriority popping up in the JobAttributes, but I haven't been able to set it in the API even using some rather creative python. How is this meant to work ?

Cheers,
Daniela

**Comment by fstagni** [comment, marked as answer]
I think it's possible with 

```
jobObject = Job()  # I guess at some point you have this
...
jobObject._addParameter(jobObject.workflow,  "Priority", "JDL", str(self.priority), "UserPriority")
```
But TBH it would be better to add a method to `API.Job`. Feel free to add it.



Another possibility is to create a new users' group, name it e.g. "high_priority_users", that has the same description of your standard users' group with the exception of the "JobShare" parameter, to which you would give a pretty higher number (yes, I know "what does that mean?" but there's no doc whatsoever, this is really ancient code).
If you go for this solution, your user would need to do `dirac-proxy-init -g high_priority_users` before submitting their jobs.

**Comment by marianne013** [comment]
I think the use case here is more per job type than per user (it's not particle physics, but something like reco > mc > analysis (and just before Moriond the reverse ;-))
